### PR TITLE
Adding my handwired keyboard and keymap

### PR DESCRIPTION
**Pull request by obuwunkunubi**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+  *      | NO| 0 | . |ENT|
+  *      `---------------'
+  */
+  [BASE] = LAYOUT_spaget_ortho_6x4(     \
+    KC_CALC, TO(ONE), KC_MPRV, KC_MNXT, \
+    KC_EQL,  KC_PSLS, KC_PAST, KC_MINS, \
+    KC_P7,   KC_P8,   KC_P9,   KC_NLCK, \
+    KC_P4,   KC_P5,   KC_P6,   KC_PPLS, \
+    KC_P1,   KC_P2,   KC_P3,   KC_MPLY, \
+    KC_NO,   KC_P0,   KC_PDOT, KC_PENT  \
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:79]:
```suggestion
             KC_P0,   KC_PDOT, KC_PENT  \
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+  *      | 7 | 8 | 9 |NUM|--  |
+  *      |---------------|    |
+  *      | 4 | 5 | 6 | + |    |
+  *      |---------------|    |
+  *      | 1 | 2 | 3 |PLY|-----
+  *      |---------------|
+  *      | NO| 0 | . |ENT|
+  *      `---------------'
+  */
+  [BASE] = LAYOUT_spaget_ortho_6x4(     \
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:73]:
```suggestion
  [BASE] = LAYOUT(     \
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+  *      |SWP|AF4|SWN|RST|--  |
+  *      |---------------|    |
+  *      |CPY| U |PST|PSC|    |
+  *      |---------------|    |
+  *      | L | D | R |CAD|-----
+  *      |---------------|
+  *      | NO| NO| NO| NO|
+  *      `---------------'
+  */
+  [ONE] = LAYOUT_spaget_ortho_6x4(       \
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:103]:
```suggestion
  [ONE] = LAYOUT(       \
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+    TO(BASE), TO(TWO), MAKE1,   MAKE2,   \
+    DIR,      MAIL1,   MAIL2,   OBUWUN,  \
+    SELWP,    KC_AF4,  SELWN,   RESET,   \
+    KC_CPY,   KC_UP,   KC_PST,  KC_PSCR, \
+    KC_LEFT,  KC_DOWN, KC_RGHT, KC_CAD,  \
+    KC_NO,    KC_P0,   KC_PDOT, KC_PENT  \
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:109]:
```suggestion
             KC_P0,   KC_PDOT, KC_PENT  \
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+  *      |EM7|EM8|EM9| NO|--  |
+  *      |---------------|    |
+  *      |EM4|EM5|EM6| NO|    |
+  *      |---------------|    |
+  *      |EM1|EM2|EM3| NO|-----
+  *      |---------------|
+  *      | NO| NO| NO| NO|
+  *      `---------------'
+  */
+  [TWO] = LAYOUT_spaget_ortho_6x4(       \
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:134]:
```suggestion
  [TWO] = LAYOUT(       \
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+    TO(ONE), TO(BASE), KC_LBRC, KC_RBRC, \
+    KC_F13,  KC_F14,   KC_F15,  KC_F16,  \
+    EM7,     EM8,      EM9,     KC_NO,   \
+    EM4,     EM5,      EM6,     KC_NO,   \
+    EM1,     EM2,      EM3,     KC_NO,   \
+    KC_NO,   KC_NO,    KC_NO,   KC_NO    \
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:140]:
```suggestion
             KC_NO,    KC_NO,   KC_NO    \
```

File: keyboards/handwired/obuwunkunubi/spaget/info.json [hunk 7]
```diff
@@ -0,0 +1,37 @@
+{
+    "keyboard_name": "spaget",
+    "url": "",
+    "maintainer": "obuwunkunubi",
+    "width": 4,
+    "height": 6,
+    "layouts": {
+        "LAYOUT_spaget_ortho_6x4": {
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/info.json:8]:
```suggestion
        "LAYOUT": {
```

File: keyboards/handwired/obuwunkunubi/spaget/info.json [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+				{"x":0, "y":3},
+				{"x":1, "y":3},
+				{"x":2, "y":3},
+				{"x":3, "y":3},
+				{"x":0, "y":4},
+				{"x":1, "y":4},
+				{"x":2, "y":4},
+				{"x":3, "y":4},
+        {"x":0, "y":5},
+				{"x":1, "y":5},
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/info.json:31]:
```suggestion
				{"x":0, "y":5, "w":2},
```

File: keyboards/handwired/obuwunkunubi/spaget/rules.mk [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+BACKLIGHT_ENABLE = no    # Enable keyboard backlight functionality
+MIDI_ENABLE = no         # MIDI controls
+AUDIO_ENABLE = no
+UNICODE_ENABLE = yes     # Unicode
+BLUETOOTH_ENABLE = no    # Enable Bluetooth with the Adafruit EZ-Key HID
+
+OLED_DRIVER_ENABLE = yes # Enable OLED display support
+ENCODER_ENABLE = yes	   # Enable encoder support
+
+LAYOUTS = spaget_ortho_6x4
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/rules.mk:33]:
```suggestion
```

File: keyboards/handwired/obuwunkunubi/spaget/spaget.h [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+// readability
+#define XXX KC_NO
+
+#define LAYOUT_spaget_ortho_6x4( \
+	K00, K01, K02, K03, \
+	K10, K11, K12, K13, \
+	K20, K21, K22, K23, \
+	K30, K31, K32, K33, \
+	K40, K41, K42, K43, \
+  XXX, K51, K52, K53  \
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/spaget.h:38]:
```suggestion
	     K51, K52, K53  \
```

File: keyboards/handwired/obuwunkunubi/spaget/spaget.h [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+ * layout of the board and position of the keys.
+ *
+ * The second converts the arguments into a two-dimensional array which
+ * represents the switch matrix.
+ */
+
+// readability
+#define XXX KC_NO
+
+#define LAYOUT_spaget_ortho_6x4( \
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/spaget.h:32]:
```suggestion
#define LAYOUT( \
```

File: keyboards/handwired/obuwunkunubi/spaget/spaget.c [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+  // put your keyboard start-up code here
+  // runs once when the firmware starts up
+  matrix_init_user();
+};
+
+void matrix_scan_kb(void) {
+  // put your looping keyboard code here
+  // runs every cycle (a lot)
+  matrix_scan_user();
+};
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/spaget.c:28]:
```suggestion
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/rules.mk [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = yes        # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+
+OLED_DRIVER_ENABLE = yes		# Enable OLED display
+ENCODER_ENABLE = yes				# Enable encoder support
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/rules.mk:20]: File should only contain overrides, which this looks like none. I would suggest this file is deleted or...
```suggestion
# Build overrides
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 14]
```diff
@@ -0,0 +1,446 @@
...
+};
+
+void matrix_scan_user(void) {
+
+}
+
+void led_set_user(uint8_t usb_led) {
+
+}
+
```
Comment (zvecr) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:354]:
```suggestion

```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 15]
```diff
@@ -0,0 +1,446 @@
...
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+
+    case SELWN:
+    if (record->event.pressed) {
+      // when keycode SELWN is pressed
+
+      send_string(SS_DOWN(X_LCTRL)SS_DOWN(X_LSHIFT)SS_TAP(X_RIGHT)SS_UP(X_LSHIFT)"c"SS_UP(X_LCTRL)); // select and copy next word
```
Comment (fauxpark) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:153]:
```suggestion
      tap_code16(C(S(KC_RGHT))); // select and copy next word
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 16]
```diff
@@ -0,0 +1,446 @@
...
+    } else {
+      // when keycode SELWN is released
+    }
+    break;
+
+    case SELWP:
+    if (record->event.pressed) {
+      // when keycode SELWP is pressed
+
+      send_string(SS_DOWN(X_LCTRL)SS_DOWN(X_LSHIFT)SS_TAP(X_LEFT)SS_UP(X_LSHIFT)"c"SS_UP(X_LCTRL)); // select and copy previous word
```
Comment (fauxpark) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:165]:
```suggestion
      tap_code16(C(S(KC_LEFT))); // select and copy previous word
```

File: keyboards/handwired/obuwunkunubi/spaget/readme.md [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+![spaget](https://imgur.com/a/pkr8Phk)
+
+A hand wired numpad style QMK macropad, with a 128x32 OLED screen and two rotary encoders. Powered by an Arduino Pro Micro.
+
+* Keyboard Maintainer: [obuwunkunubi](https://github.com/obuwunkunubi)
+* Hardware Supported: spaget v1
+
+Make example for this keyboard (after setting up your build environment):
+
+    make handwired/obuwunkunubi/spaget:default:avrdude
```
Comment (fauxpark) [inline keyboards/handwired/obuwunkunubi/spaget/readme.md:12]:
```suggestion
    make handwired/obuwunkunubi/spaget:default
```

File: keyboards/handwired/obuwunkunubi/spaget/readme.md [hunk 18]
```diff
@@ -0,0 +1,14 @@
+# spaget
+
+![spaget](https://imgur.com/a/pkr8Phk)
```
Comment (fauxpark) [inline keyboards/handwired/obuwunkunubi/spaget/readme.md:3]:
```suggestion
![spaget](https://i.imgur.com/ktlhskZ.jpg)
```

File: keyboards/handwired/obuwunkunubi/spaget/spaget.h [hunk 19]
```diff
@@ -0,0 +1,46 @@
...
+// readability
+#define XXX KC_NO
+
+#define LAYOUT( \
+	K00, K01, K02, K03, \
+	K10, K11, K12, K13, \
+	K20, K21, K22, K23, \
+	K30, K31, K32, K33, \
+	K40, K41, K42, K43, \
+	     K51, K52, K53  \
```
Comment (noroadsleft) [inline keyboards/handwired/obuwunkunubi/spaget/spaget.h:38]: The layout macro should be laid out such that the keycodes in `keymap.c` are roughly in their geographic locations, with respect to the assembled hardware.

```suggestion
  K23,           K43, \
	K00, K01, K02, K03, \
	K10, K11, K12, K13, \
	K20, K21, K22,      \
	K30, K31, K32, K33, \
	K40, K41, K42,      \
	     K51, K52, K53  \
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 20]
```diff
@@ -0,0 +1,439 @@
...
+  *      | NO| 0 | . |ENT|
+  *      `---------------'
+  */
+  [BASE] = LAYOUT(     \
+    KC_CALC, TO(ONE), KC_MPRV, KC_MNXT, \
+    KC_EQL,  KC_PSLS, KC_PAST, KC_MINS, \
+    KC_P7,   KC_P8,   KC_P9,   KC_NLCK, \
+    KC_P4,   KC_P5,   KC_P6,   KC_PPLS, \
+    KC_P1,   KC_P2,   KC_P3,   KC_MPLY, \
+             KC_P0,   KC_PDOT, KC_PENT  \
```
Comment (noroadsleft) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:79]:
```suggestion
    KC_NLCK,                   KC_MPLY, \
    KC_CALC, TO(ONE), KC_MPRV, KC_MNXT, \
    KC_EQL,  KC_PSLS, KC_PAST, KC_MINS, \
    KC_P7,   KC_P8,   KC_P9,            \
    KC_P4,   KC_P5,   KC_P6,   KC_PPLS, \
    KC_P1,   KC_P2,   KC_P3,            \
             KC_P0,   KC_PDOT, KC_PENT  \
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 21]
```diff
@@ -0,0 +1,439 @@
...
+  *      | NO| NO| NO| NO|
+  *      `---------------'
+  */
+  [ONE] = LAYOUT(       \
+    TO(BASE), TO(TWO), MAKE1,   MAKE2,   \
+    DIR,      MAIL1,   MAIL2,   OBUWUN,  \
+    SELWP,    KC_AF4,  SELWN,   RESET,   \
+    KC_CPY,   KC_UP,   KC_PST,  KC_PSCR, \
+    KC_LEFT,  KC_DOWN, KC_RGHT, KC_CAD,  \
+             KC_P0,   KC_PDOT, KC_PENT  \
```
Comment (noroadsleft) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:109]:
```suggestion
    RESET,                      KC_CAD,  \
    TO(BASE), TO(TWO), MAKE1,   MAKE2,   \
    DIR,      MAIL1,   MAIL2,   OBUWUN,  \
    SELWP,    KC_AF4,  SELWN,            \
    KC_CPY,   KC_UP,   KC_PST,  KC_PSCR, \
    KC_LEFT,  KC_DOWN, KC_RGHT,          \
             KC_P0,   KC_PDOT, KC_PENT   \
```

File: keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c [hunk 22]
```diff
@@ -0,0 +1,439 @@
...
+  *      | NO| NO| NO| NO|
+  *      `---------------'
+  */
+  [TWO] = LAYOUT(       \
+    TO(ONE), TO(BASE), KC_LBRC, KC_RBRC, \
+    KC_F13,  KC_F14,   KC_F15,  KC_F16,  \
+    EM7,     EM8,      EM9,     KC_NO,   \
+    EM4,     EM5,      EM6,     KC_NO,   \
+    EM1,     EM2,      EM3,     KC_NO,   \
+             KC_NO,    KC_NO,   KC_NO    \
```
Comment (noroadsleft) [inline keyboards/handwired/obuwunkunubi/spaget/keymaps/default/keymap.c:140]:
```suggestion
    KC_NO,                      KC_NO,   \
    TO(ONE), TO(BASE), KC_LBRC, KC_RBRC, \
    KC_F13,  KC_F14,   KC_F15,  KC_F16,  \
    EM7,     EM8,      EM9,              \
    EM4,     EM5,      EM6,     KC_NO,   \
    EM1,     EM2,      EM3,              \
             KC_NO,    KC_NO,   KC_NO    \
```

File: keyboards/handwired/obuwunkunubi/spaget/info.json [hunk 23]
```diff
@@ -0,0 +1,36 @@
+{
+    "keyboard_name": "spaget",
+    "url": "",
+    "maintainer": "obuwunkunubi",
+    "width": 4,
+    "height": 6,
```
Comment (noroadsleft) [inline keyboards/handwired/obuwunkunubi/spaget/info.json:6]:
```suggestion
    "height": 7.5,
```

File: keyboards/handwired/obuwunkunubi/spaget/info.json [hunk 24]
```diff
@@ -0,0 +1,36 @@
...
+				{"x":1, "y":3},
+				{"x":2, "y":3},
+				{"x":3, "y":3},
+				{"x":0, "y":4},
+				{"x":1, "y":4},
+				{"x":2, "y":4},
+				{"x":3, "y":4},
+				{"x":0, "y":5, "w":2},
+				{"x":2, "y":5},
+				{"x":3, "y":5}
```
Comment (noroadsleft) [inline keyboards/handwired/obuwunkunubi/spaget/info.json:32]:
```suggestion
                {"x":0, "y":0},
                {"x":3, "y":0},
                {"x":0, "y":1.5},
                {"x":1, "y":1.5},
                {"x":2, "y":1.5},
                {"x":3, "y":1.5},
                {"x":0, "y":2.5},
                {"x":1, "y":2.5},
                {"x":2, "y":2.5},
                {"x":3, "y":2.5},
                {"x":0, "y":3.5},
                {"x":1, "y":3.5},
                {"x":2, "y":3.5},
                {"x":0, "y":4.5},
                {"x":1, "y":4.5},
                {"x":2, "y":4.5},
                {"x":3, "y":3.5, "h":2},
                {"x":0, "y":5.5},
                {"x":1, "y":5.5},
                {"x":2, "y":5.5},
                {"x":0, "y":6.5, "w":2},
                {"x":2, "y":6.5},
                {"x":3, "y":5.5, "h":2}
```

File: keyboards/handwired/obuwunkunubi/spaget/spaget.h (hunk 19 above)
Comment (obuwunkunubi) [reply keyboards/handwired/obuwunkunubi/spaget/spaget.h:38]: Thanks for the info. I wanted to do something like this initially, but I wasn't quite sure how to. I will make the changes asap.

**Comment by noroadsleft** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks!

**Pull request merged**